src/material-experimental/mdc-slider/slider.ts

```diff
@@ -94,13 +94,13 @@ export class MatSliderVisualThumb implements AfterViewInit, OnDestroy {
   private _sliderInput: MatSliderThumb;
 
   /** The RippleRef for the slider thumbs hover state. */
-  private _hoverRippleRef: RippleRef;
+  _hoverRippleRef: RippleRef;
```

Comment:
Can we check these by selector instead? I don't like when we have to make a bunch of private stuff fake private just for tests

Reply:
@mmalerba How does this sound?

#### Proposed solution

```ts
function isRippleVisible(selector: string) {
      tick(500);
      return !!document.querySelector(`.mat-mdc-slider-${selector}-ripple`);
}
```

And wrap all the tests in `fakeAsync`.

```ts
it('should show the hover ripple on mouseenter', fakeAsync(() => {
      expect(isRippleVisible('hover')).toBe(false);
      mouseenter();
      expect(isRippleVisible('hover')).toBe(true);
}));
```

We need the `tick(500)` because ripples that are in the fading out state are still in the DOM, so we need to wait till they're completely hidden before querying.

Reply:
yeah that looks good